### PR TITLE
[1]: Authentication

### LINKED ISSUE #1: [DMP 2024]: What's your Ripple Effect?
# Ticket Contents

# Description
According to studies, if we keep using water the way we do today, in 16 years we will have no freshwater left on Earth. Apart from the direct water use that we have in the way of household activities, there is a massive indirect water use that is not obvious to us. This use happens through the goods we consume.

Create a mobile/responsive water footprint calculator. The calculator allows users to select a dish from a list. Once a dish is selected, a user will see nutritional information and the water footprint of the dish

# Goals & Mid-Point Milestone

## Goals

- [x] Plan the architecture, design and structure of the repository.
- [x] Configure Backend Server with PostgreSQL DB
- [x] Analyze the Dataset, populate it and generated it as per the requirement
- [x] Implement Authentication (Sign Up, login and Forgot Password)
- [x] Implement Home Page UI
- [x] Work on Water Calculator logic.
- [x] Implement Additional functionalities (ex: quiz, discovery page, etc.)

# Setup/Installation
Android Studio
VS Code
PgAdmin

# Expected Outcome
A mobile (PWA) app that users can interact with. It will allow users to
Calculate their water footprint by entering the food they consume
Track their water footprint
Share their water footprint chart on social media

# Acceptance Criteria
No response

# Implementation Details
Any other frameworks can be used
NextJS
NodeJS
MySQL

# Mockups/Wireframes
[Puddles](https://www.nishthaluhadia.com/puddle)

[View in Drive](https://drive.google.com/file/d/1o50g33tkZd1bspvnDLBh7dIFz3xXuGbn/view)

# Product Name
India Water Portal

# Organisation Name
Arghyam

# Domain
Water

# Tech Skills Needed
Android, Kotlin, Jetpack Compose, Node.js, Typescript, PostgresSql

# Mentor(s)
Sreechand Tavva, Simran Nigam, Arpan Kumar De

# Category
Backend, Frontend, Mobile

### PR DESCRIPTION
**Pull request by kevalkanp1011**
#1 

Authentication APIs Implemented as below

1. **Login API**
route: "/login"
_**Request Schema:**_
username: string, password: string

2. **Register API**
route: "/register"
_**Request Schema:**_
name: string, username: string, password: string

3. **Send Forgot Password Otp API**
route: "/generate-otp"
_**Request Schema:**_
email: string

4. **Verify Forgot Password Otp API**
route: "/verify-otp"
 _**Request Schema:**_
email: string
otp: string
created_on: number

6. **Reset Password API**
route: "/reset-password"
 _**Request Schema:**_
email: string
new_password: string
access_token: string

### CODE REVIEW

**Comment by arpankumarde** [comment]
Could you please add a `.gitignore` file to exclude unnecessary files and folders from being committed?

**Comment by kevalkanp1011** [comment]
Okay I'll do

**Comment by kevalkanp1011** [comment]
I have added the .gitignore file by removing unnecessary files (dist and node_modules)

File: db.ts
Comment (arpankumarde) [inline db.ts:1]: Create a new environment variable for storing the Connection string and then use it here

File: domain/models/Otp.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  declare id: number;
+  declare email: string;
+  declare otp_hash: string;
+  declare generated_at: Date;
+
+}
+
+export function initOtp(sequelize: Sequelize) {
+  
+    console.log("Otp models were synchronized successfully.");
```
Comment (simran142002) [inline domain/models/Otp.ts]: remove the consoles

Comment (kevalkanp1011) [reply domain/models/Otp.ts]: I've removed it.

**Review by arpankumarde — APPROVED** [review]
Looks good. I'm merging it

**Pull request merged**